src/miner.cpp

```diff
@@ -86,17 +80,16 @@ static BlockAssembler::Options DefaultOptions(const CChainParams& params)
     // If both are given, restrict both.
     BlockAssembler::Options options;
     options.nBlockMaxWeight = DEFAULT_BLOCK_MAX_WEIGHT;
-    options.nBlockMaxSize = DEFAULT_BLOCK_MAX_SIZE;
     bool fWeightSet = false;
     if (gArgs.IsArgSet("-blockmaxweight")) {
         options.nBlockMaxWeight = gArgs.GetArg("-blockmaxweight", DEFAULT_BLOCK_MAX_WEIGHT);
-        options.nBlockMaxSize = MAX_BLOCK_SERIALIZED_SIZE;
         fWeightSet = true;
     }
     if (gArgs.IsArgSet("-blockmaxsize")) {
-        options.nBlockMaxSize = gArgs.GetArg("-blockmaxsize", DEFAULT_BLOCK_MAX_SIZE);
         if (!fWeightSet) {
-            options.nBlockMaxWeight = options.nBlockMaxSize * WITNESS_SCALE_FACTOR;
+            options.nBlockMaxWeight = gArgs.GetArg("-blockmaxsize", DEFAULT_BLOCK_MAX_WEIGHT / WITNESS_SCALE_FACTOR) * WITNESS_SCALE_FACTOR;
```

Comment:
Add warn about deprecation?

Reply:
Agree. A deprecation warning here would be helpful.

Reply:
I'd prefer to not add more garbage to init, if possible (and printing every time a BlockAssembler is created also sucks, though it will do that now if you set both :( ). I think its fine, and since we're changing the defaults to something reasonable, removing the option a release later likely won't break anything.

Reply:
Then add the `LogPrintf()`s to `InitParameterInteraction()` in init.cpp instead.

Reply:
Ugh, fine.